### PR TITLE
add auto create vendor feature

### PR DESCRIPTION
**Pull request by Hrishabh17**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new configuration option to automatically create merchants as vendors in the application.
	- Enhanced vendor management capabilities with new parameters influencing vendor creation based on specific criteria.
	- Added a method to sanitize vendor names for improved data integrity.

- **Bug Fixes**
	- Improved error handling during vendor retrieval processes.

- **Tests**
	- Updated tests to incorporate the new configuration settings and validate vendor creation logic.
	- Added tests for the new vendor name sanitization method.

### CODE REVIEW

File: tests/test_sageintacct/test_utils.py [hunk 1]
```diff
@@ -760,3 +760,73 @@ def test_get_or_create_employee(mocker, db):
...
+
+    # Test case 11: Vendor name that is an empty string
+    vendor_name = ""
+    expected_output = None
+    assert sage_intacct_connection.sanitize_vendor_name(vendor_name) == expected_output
+
+    # Test case 12: Vendor name with tabs and newlines
+    vendor_name = "Vendor\tName\n123"
+    expected_output = "Vendor Name 123"
+    assert sage_intacct_connection.sanitize_vendor_name(vendor_name) == expected_output
```
Comment (ashwin1111) [inline tests/test_sageintacct/test_utils.py:832]: good stuff with detailed cases

**Review by ashwin1111 — APPROVED** [review]

File: apps/sage_intacct/import_helpers.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import logging
```
Comment (ashwin1111) [inline apps/sage_intacct/import_helpers.py:1]: move to helpers file

Comment (ashwin1111) [reply apps/sage_intacct/import_helpers.py:1]: no need of separate file

File: apps/sage_intacct/import_helpers.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+def get_or_create_credit_card_vendor(workspace_id: int, configuration: Configuration, merchant: str = None):
+    """
+    Get or create default vendor
+    :param merchant: Fyle Expense Merchant
+    :param workspace_id: Workspace Id
+    :return:
+    """
+    sage_intacct_credentials = SageIntacctCredential.objects.get(workspace_id=workspace_id)
+    sage_intacct_connection = import_string('apps.sage_intacct.utils.SageIntacctConnector')(sage_intacct_credentials, workspace_id)
```
Comment (ashwin1111) [inline apps/sage_intacct/import_helpers.py:20]: let's init sdk in parent func and pass here, this shouldn't initialised for every expense

File: apps/sage_intacct/models.py [hunk 4]
```diff
@@ -934,8 +935,9 @@ def create_journal_entry_lineitems(expense_group: ExpenseGroup, configuration: C
 
                 vendor_id = entity.destination_vendor.destination_id if employee_mapping_setting == 'VENDOR' else None
 
-                if lineitem.fund_source == 'CCC' and configuration.use_merchant_in_journal_line and lineitem.vendor:
-                    vendor = DestinationAttribute.objects.filter(attribute_type='VENDOR', value__iexact=lineitem.vendor, workspace_id=expense_group.workspace_id).order_by('-updated_at').first()
+                if lineitem.fund_source == 'CCC' and configuration.use_merchant_in_journal_line:
+                    # here it would create a Credit Card Vendor if the expene vendor is not present
+                    vendor = get_or_create_credit_card_vendor(expense_group.workspace_id, configuration, lineitem.vendor)
```
Comment (ashwin1111) [inline apps/sage_intacct/models.py:940]: let's check DB first, if it doesn't exist can call get_or_create_credit_card_vendor() - this is to optimise number of calls to Intacct

Comment (Hrishabh17) [reply apps/sage_intacct/models.py:940]: we are not directly calling the intacct, inside the sage connector -> get_or_create_vendor, we are first checking db and returning if found.

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**